oidc-authentication.md

```diff
@@ -128,10 +128,139 @@ WWW-Authenticate: DPoP realm=”https://example.com” scope=”openid webid”,
 
 # 8. Authenticated Requests
 
-## 8.1 Use of `Authorization` header
+When making authenticated requests to [Resource Server]()s, a [Client]() uses
+a combination of the `Authorization` header and an appropriate Proof of
+Possession header (such as DPoP).
+
+## 8.1 Use of Authorization header
+
+### For Multi-RS / “Social App” Use Cases
+
+For Multi-RS use cases, a client MUST use an `Authorization` header as well
+as the `DPoP` proof-of-possession header.
+
+**Authorization:** `DPoP <include JWT-encoded ID Credential here>`<br/>
+**DPoP:** `<include a sender-constrained DPoP Proof JWT (see example below)>`
+
+Example authenticated request:
+
+```
+GET /protectedresource HTTP/1.1
+ Host: resource.example.org
+ Authorization: DPoP eyJhbGciOiJFUzI1NiIsImtpZCI6IkJlQUxrYiJ9.eyJzdWI
+  iOiJzb21lb25lQGV4YW1wbGUuY29tIiwiaXNzIjoiaHR0cHM6Ly9zZXJ2ZXIuZXhhbX
+  BsZS5jb20iLCJhdWQiOiJodHRwczovL3Jlc291cmNlLmV4YW1wbGUub3JnIiwibmJmI
+  joxNTYyMjYyNjExLCJleHAiOjE1NjIyNjYyMTYsImNuZiI6eyJqa3QiOiIwWmNPQ09S
+  Wk5ZeS1EV3BxcTMwalp5SkdIVE4wZDJIZ2xCVjN1aWd1QTRJIn19.vsFiVqHCyIkBYu
+  50c69bmPJsj8qYlsXfuC6nZcLl8YYRNOhqMuRXu6oSZHe2dGZY0ODNaGg1cg-kVigzY
+  hF1MQ
+ DPoP: eyJ0eXAiOiJkcG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6Ik
+  VDIiwieCI6Imw4dEZyaHgtMzR0VjNoUklDUkRZOXpDa0RscEJoRjQyVVFVZldWQVdCR
+  nMiLCJ5IjoiOVZFNGpmX09rX282NHpiVFRsY3VOSmFqSG10NnY5VERWclUwQ2R2R1JE
+  QSIsImNydiI6IlAtMjU2In19.eyJqdGkiOiJlMWozVl9iS2ljOC1MQUVCIiwiaHRtIj
+  oiR0VUIiwiaHR1IjoiaHR0cHM6Ly9yZXNvdXJjZS5leGFtcGxlLm9yZy9wcm90ZWN0Z
+  WRyZXNvdXJjZSIsImlhdCI6MTU2MjI2MjYxOH0.lNhmpAX1WwmpBvwhok4E74kWCiGB
+  NdavjLAeevGy32H3dbF0Jbri69Nm2ukkwb-uyUI4AUg1JSskfWIyo4UCbQ
+```
+
+Example contents of `Authorization` header (this is the ID Credential in its
+entirety, the contents of the `id_vc` claim from the IDP's ID Token):
+
+```
+{
+  "sub": "https://janedoe.com/web#id",    <- Web ID / DID
+  "iss": "https://idp.example.com",
+  "aud": "https://client.example.com", // audience constrained to the client / presenter
+  "iat": 1541493724,
+  "exp": 1573029723,  <- identity credential expiration
+  "cnf":{
+      // DPoP public key confirmation, per DPoP spec section 7
+      "jkt":"0ZcOCORZNYy-DWpqq30jZyJGHTN0d2HglBV3uiguA4I"  
+  }
+}
+// This ID Credential is signed by the IDP, NOT by the client
+```
+
+Example DPoP Proof JWT (minted by Client app for each resource) (the
+contents of the `DPoP` header):
+
+```js
+{
+   "typ":"dpop+jwt",
+   "alg":"ES256",
+   "jwk": {   // <- gets checked against the cnf claim in the ID Credential (from the Authorization header)
+     "kty":"EC",
+     "x":"l8tFrhx-34tV3hRICRDY9zCkDlpBhF42UQUfWVAWBFs",
+     "y":"9VE4jf_Ok_o64zbTTlcuNJajHmt6v9TDVrU0CdvGRDA",
+     "crv":"P-256"
+   }
+ }.{
+   "jti":"-BwC3ESc6acc2lTc",
+   "iat":1562262616,
+    // the following fields are required only for the authenticated resource request (not for the token endpoint)
```

Comment:
This is a departure from the DPoP specification. In a spec-compliant DPoP interaction, a DPoP header is also be sent to the token endpoint, and in order for that DPoP header to be valid, it must contain the `htm` and `htu` claims.

See, for instance, [Section 5](https://tools.ietf.org/html/draft-fett-oauth-dpop-03#section-5) of the DPoP spec:

> To bind a token to a public key in the token request, the client MUST provide a valid DPoP proof JWT in a "DPoP" header.

And per [section 4.2](https://tools.ietf.org/html/draft-fett-oauth-dpop-03#section-4.2) a valid DPoP proof will pass checks for:

>    6.  the "htm" claim matches the HTTP method value of the HTTP request in which the JWT was received (case-insensitive),
>
> 7.  the "htu" claims matches the HTTP URI value for the HTTP request in which the JWT was received, ignoring any query and fragment parts,

Sending a DPoP header to the token endpoint is a vitally important step in this interaction; otherwise, how does the confirmation claim make its way into the resulting JWT? And that claim is validated, in part, by ensuring that it has been created with the _particular_ token endpoint in mind.

Reply:
Not quite a departure, although I think I need to rephrase that comment, since it's confusing.
What I'm referring to is [Section 6 - Resource Access](https://tools.ietf.org/html/draft-fett-oauth-dpop-03#section-6). (It just so happens that a similar proof is _also_ done at the token endpoint, but that's described in another session of the spec.)

Reply:
Great, as long as we're clear that the same DPoP proofing happens at the token endpoint